File: CS/MvcDashboard_ServerExport/Controllers/HomeController.cs
using DevExpress.DashboardCommon;
using DevExpress.DashboardWeb;
using System;
using System.IO;
using System.Web.Mvc;

namespace MvcDashboard_ServerExport.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }

        [HttpPost]
        public ActionResult ExportDashboardToPdf(string DashboardID, string DashboardState) {
            using (MemoryStream stream = new MemoryStream()) {
                string dashboardID = DashboardID;
                DashboardState dashboardState = new DashboardState();
                dashboardState.LoadFromJson(DashboardState);

                DashboardPdfExportOptions pdfOptions = new DashboardPdfExportOptions();
                pdfOptions.ExportFilters = true;
                pdfOptions.DashboardStatePosition = DashboardStateExportPosition.Below;

                string dateTimeNow = DateTime.Now.ToString("yyyyMMddHHmmss");
                string filePath = "~/App_Data/Export/" + dashboardID + "_" + dateTimeNow + ".pdf";
                ASPxDashboardExporter exporter = new ASPxDashboardExporter(DashboardConfigurator.Default);
                exporter.ExportToPdf(dashboardID, stream, new System.Drawing.Size(1024, 768), dashboardState, pdfOptions);
                SaveFile(stream, filePath);

                ContentResult result = new ContentResult();
                result.Content = filePath;
                return result;
            }
        }

        private void SaveFile(MemoryStream stream, string path) {
            var fileStream = System.IO.File.Create(Server.MapPath(path));
            stream.WriteTo(fileStream);
            fileStream.Close();
        }
    }
}

File: VB/MvcDashboard_ServerExport/Controllers/HomeController.vb
Imports System.IO
Imports System.Web.Mvc
Imports DevExpress.DashboardCommon
Imports DevExpress.DashboardWeb

Namespace MvcDashboard_ServerExport.Controllers
    Public Class HomeController
        Inherits Controller

        Public Function Index() As ActionResult
            Return View()
        End Function

		<HttpPost>
		Public Function ExportDashboardToPdf(ByVal DashboardID As String, ByVal DashboardState As String) As ActionResult
			Using stream As New MemoryStream()
				Dim dbID As String = DashboardID
				Dim dbState As New DashboardState()

				dbState.LoadFromJson(DashboardState)
				Dim pdfOptions As New DashboardPdfExportOptions()
				pdfOptions.ExportFilters = True
				pdfOptions.DashboardStatePosition = DashboardStateExportPosition.Below
				Dim dateTimeNow As String = Date.Now.ToString("yyyyMMddHHmmss")
				Dim serverPath As String = Server.MapPath("~/App_Data/Export")
				If Not Directory.Exists(serverPath) Then
					Directory.CreateDirectory(serverPath)
				End If
				Dim filePath As String = Path.Combine(serverPath, dbID)
				Dim uniqueId As String = "_" & dateTimeNow & ".pdf"
				Dim exporter = New WebDashboardExporter(DashboardConfigurator.Default)
				exporter.ExportToPdf(dbID, stream, New System.Drawing.Size(1024, 768), dbState, pdfOptions)
				SaveFile(stream, filePath & uniqueId)
				Dim result As New ContentResult()
				result.Content = filePath & uniqueId
				Return result
			End Using
		End Function

		Private Sub SaveFile(ByVal stream As MemoryStream, ByVal path As String)
			Dim fileStream = System.IO.File.Create(path)
			stream.WriteTo(fileStream)
            fileStream.Close()
        End Sub
    End Class
End Namespace